src/radon.ts

```diff
@@ -32,14 +40,29 @@ export class Radon {
     ;(this.context.cache.get(scriptId) as Script).addOperator()
   }
 
+  public onChildrenEvent() {
+    return {
+      emit: (e: { sourceType: Kind }) => {
+        this.sourceType = e.sourceType
+        if (this.sourceType === Kind.RNG) {
```

Comment:
```suggestion
        if (this.kind === Kind.RNG) {
```

Reply:
It makes sense to name the type of the source as `kind` in the `source.ts` file. In other files the naming should be more explicative. I think, in this case, sourceType is a better name.